File: benchmark.ts
const { performance } = require("perf_hooks");
const os = require("os");

const start = performance.now();
let sum = 0;

for (let i = 1; i < 1000000001; ++i) {
  sum += i;
}

const end = performance.now();

console.log(`Sum: ${sum}`);
console.log(`Time taken: ${end - start} ms`);

File: benchmark.cpp
#include <chrono>
#include <iostream>

int main() {
  auto start = std::chrono::high_resolution_clock::now();
  long long sum = 0;

  for (long long i = 1; i < 1000000001; ++i) {
    sum += i;
  }

  auto end = std::chrono::high_resolution_clock::now();
  auto duration =
      std::chrono::duration_cast<std::chrono::milliseconds>(end - start)
          .count();

  std::cout << "Sum: " << sum << "\n";
  std::cout << "Time taken: " << duration << " ms" << "\n";
}
